src/codegen/error.rs

```diff
@@ -0,0 +1,36 @@
+use std::error;
+use std::fmt;
+
+/// Errors that can occur during code generation.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Error {
```

Comment:
This seems like an overly generic name for a really specific error kind. I don't have a clearly better name in mind though.